### PR TITLE
tests(share/byzantine): extend befp tests

### PR DESCRIPTION
**Pull request by vgonkivs**
## Overview
Extend test cases for befp.Validate.

## Checklist
- [ ] New and updated code has appropriate documentation
- [ ] New and updated code has new and/or updated testing
- [ ] Required CI checks are passing
- [ ] Visual proof for any user facing features like CLI or documentation updates
- [ ] Linked issues closed with keywords

### CODE REVIEW

File: share/eds/byzantine/bad_encoding_test.go [hunk 1]
```diff
@@ -37,10 +38,92 @@ func TestBadEncodingFraudProof(t *testing.T) {
...
+			doFn: func() error {
+				befp, ok := befp.(*BadEncodingProof)
+				require.True(t, ok)
+				befpCopy := *befp
+				befpCopy.Index = 100
+				return befpCopy.Validate(&header.ExtendedHeader{DAH: &dah})
+			},
+			expectedResult: func(err error) {
+				require.Error(t, err)
+			},
```
Comment (renaynay) [inline share/eds/byzantine/bad_encoding_test.go:81]: Do you want to assert certain things about the error too?

**Review by renaynay — APPROVED** [review]
one nit (that applies to the table tests that expect an err)

File: share/eds/byzantine/bad_encoding_test.go (hunk 1 above)
Comment (vgonkivs) [reply share/eds/byzantine/bad_encoding_test.go:81]: I don't think so. We don't have specific errors in `Validate`

**Review by distractedm1nd — DISMISSED** [review]

File: share/eds/byzantine/bad_encoding_test.go [hunk 2]
```diff
@@ -90,3 +173,39 @@ func TestIncorrectBadEncodingFraudProof(t *testing.T) {
 	err = proof.Validate(h)
 	require.Error(t, err)
 }
+
+func TestBEFP_ValidateOutOfOrderShares(t *testing.T) {
+	// skipping it for now because `malicious` package has a small issue: Constructor does not apply
+	// passed options, so it's not possible to store shares and thus get proofs for them.
+	// should be ok once app team will fix it.
```
Comment (Wondertan) [inline share/eds/byzantine/bad_encoding_test.go:180]: Please link the issue or PR. At least in the PR description

File: share/eds/byzantine/bad_encoding_test.go [hunk 3]
```diff
@@ -37,10 +38,92 @@ func TestBadEncodingFraudProof(t *testing.T) {
...
 
 	befp := CreateBadEncodingProof([]byte("hash"), 0, errByz)
-	err = befp.Validate(&header.ExtendedHeader{
-		DAH: &dah,
-	})
-	assert.NoError(t, err)
+
+	var test = []struct {
+		name           string
+		doFn           func() error
```
Comment (Wondertan) [inline share/eds/byzantine/bad_encoding_test.go:44]: nit: calling it something like setup or prepare would make more sense

File: share/eds/byzantine/bad_encoding_test.go (hunk 1 above)
Comment (Wondertan) [reply share/eds/byzantine/bad_encoding_test.go:81]: [blocking]
We should introduce them then. When I was writing unit tests in the go-header for Verify, I first similarly omitted custom error sentinels but then saw that some tests returned errors for reasons different from the setup after code changes, which made me introduce errors. This errors do not have to be public, but they are necessary to ensure units are correct.

**Review by Wondertan — CHANGES_REQUESTED** [review]

**Review by Wondertan — APPROVED** [review]

**Review by walldiss — DISMISSED** [review]

**Review by Wondertan — APPROVED** [review]

File: share/eds/byzantine/bad_encoding_test.go (hunk 2 above)
Comment (vgonkivs) [reply share/eds/byzantine/bad_encoding_test.go:203]: Will create an issue once this PR is merged(to point the exact line of the code in main)

**Review by walldiss — DISMISSED** [review]

File: share/eds/byzantine/bad_encoding.go [hunk 4]
```diff
@@ -102,15 +102,31 @@ func (p *BadEncodingProof) UnmarshalBinary(data []byte) error {
 	return nil
 }
 
+var (
+	errHeightMismatch          = errors.New("block's height does not match with the header's height")
```
Comment (renaynay) [inline share/eds/byzantine/bad_encoding.go:106]: i'd say something like "height reported in proof" not `block's height`

**Review by renaynay — DISMISSED** [review]

**Review by Wondertan — APPROVED** [review]

**Review by renaynay — APPROVED** [review]

**Pull request merged**